### PR TITLE
feat: timezone-aware context injection for agent prompts

### LINKED ISSUE #483: Bug: User <> server timezone mismatch caused task to run at wrong time
## Description

User TZ: PST (UTC-8)
nanoclaw server TZ: UTC

* I asked for a reminder for later in the day via WhatsApp. 
* Nanoclaw said it was scheduled. 
* I received the reminder 8 hours earlier than expected.

Telling nanoclaw to remember the correct user timezone does not reliably address the issue. Perhaps this should be included as part of setup.

### LINKED ISSUE #526: Agent is repeatedly confused about the local time
I am in Sydney time zone and NanoClaw gets repeatedly confused about what time it is, despite repeated prompts and corrections. It complains about receiving message timestamps in UTC. Sometimes it think it's UTC time, and sometimes it thinks it's off by one hour. Using Opus 4.6.

### PR DESCRIPTION
**Pull request by gabi-simons**
## Summary

This PR enhances agent time-awareness by injecting a `<context>` header with the server's local time and timezone into every prompt. It eliminates agent confusion caused by raw UTC timestamps without adding database complexity or per-group configuration.

**Before** — agent receives raw UTC, cannot determine day of week or local time:
```xml
<messages>
<message sender="Alice" time="2026-02-04T18:35:22.123Z">When should we meet?</message>
</messages>
```

**After** — agent sees local time with full timezone context:
```xml
<context timezone="Africa/Johannesburg" current_time="Wednesday, February 4, 2026 8:35 PM SAST" />
<messages>
<message sender="Alice" time="Feb 4, 2026, 8:35 PM">When should we meet?</message>
</messages>
```

### Key Changes

- **Mandatory `timezone` parameter in `formatMessages()`** — every agent interaction is guaranteed to receive local time context
- **`formatLocalTime()` and `formatCurrentTime()` helpers** in `src/timezone.ts` — uses `Intl` API, zero external dependencies
- **`<context>` header injection** — timezone name + current local time in every prompt
- **Global `TIMEZONE` from config** — simple, no per-group complexity or DB changes

### Files Changed

| File | Change |
|------|--------|
| `src/timezone.ts` (NEW) | `formatLocalTime()` and `formatCurrentTime()` helpers |
| `src/router.ts` | `formatMessages()` now requires timezone, injects `<context>` header |
| `src/index.ts` | Passes `TIMEZONE` to `formatMessages()` at all call sites |
| `src/ipc.ts` | Uses `TIMEZONE` for cron/schedule parsing |
| `src/task-scheduler.ts` | Uses `TIMEZONE` for cron `next_run` computation |
| `src/timezone.test.ts` (NEW) | Unit tests for timezone utilities |
| `src/formatting.test.ts` | Updated for mandatory timezone parameter + new timezone-aware tests |

## Test Plan

- [x] `npm run build` — zero type errors
- [x] `npx vitest run` — 302/302 tests pass (30/30 test files)
- [x] Zero ghost code from prior implementation (`resolveGroupTimezone`, `set_timezone`, etc.)
- [ ] Manual: send message to group, verify agent receives `<context>` header with local time
- [ ] Manual: ask agent to schedule a task, verify correct UTC storage

## Community Credits

This PR builds on research and contributions from the community:

- **@shawny011717** ([PR #517](https://github.com/qwibitai/nanoclaw/pull/517)) — pioneered timezone-aware scheduling concepts
- **@Lafunamor** ([PR #643](https://github.com/qwibitai/nanoclaw/pull/643)) — identified `TIMEZONE` constant divergence for `once`-type tasks
- **@cachance7** ([Issue #483](https://github.com/qwibitai/nanoclaw/issues/483)) — original bug report that catalyzed this work
- **@alexdonaldsonnz** ([Issue #526](https://github.com/qwibitai/nanoclaw/issues/526)) — documented agent timezone confusion with message timestamps
- **@glifocat** ([Issue #698](https://github.com/qwibitai/nanoclaw/issues/698)) — identified agents cannot determine day of week from ISO timestamps

Their research on timezone handling guided this simplified solution. PRs #517 and #643 are superseded by this more focused approach.

Closes #483
Closes #526

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by Koshkoshinsk** [comment]
## Manual QA — Timezone PR

Fresh clone, full install (Telegram, Docker, Linux). All tests pass.

### Edge Case

Changing a group's timezone doesn't recompute `next_run` for existing cron/interval tasks — only subsequent runs use the new TZ.

### Results

| # | Test | Result |
|---|------|--------|
| A | Set timezone to America/New_York | PASS — DB `container_config` updated |
| B | Context injection (ask current time) | PASS — Agent replied "8:32 AM EST", not UTC |
| C | Once-task UTC conversion (9 AM EST) | PASS — `next_run` = `14:00:00Z` |
| D | Invalid timezone (Israel/Mars) | PASS — Rejected, DB unchanged |
| E | Switch to Asia/Tokyo, ask time | PASS — "10:38 PM JST" |
| F | Cron 9 AM (TZ=Asia/Tokyo) | PASS — `next_run` = `00:00:00Z` |
| G | No-timezone fallback | PASS — Framework fell back to UTC |
| H | DST spring-forward (2:30 AM, Mar 8) | PASS — Resolved to 3:30 AM EDT (`07:30:00Z`) |
| I | Container TZ env var | PASS — `TZ=America/New_York` |

File: src/timezone.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    .toLocaleString('en-US', {
+      timeZone: timezone,
+      timeZoneName: 'short',
+    })
+    .split(', ')
+    .pop()!
+    .split(' ')
+    .pop()!;
+  return `${formatted} ${tzAbbr}`;
+}
```
Comment (gavrielc) [inline src/timezone.ts]: I think we can drop this

File: src/router.ts [hunk 2]
```diff
@@ -9,12 +10,18 @@ export function escapeXml(s: string): string {
...
+export function formatMessages(
+  messages: NewMessage[],
+  timezone: string,
+): string {
+  const lines = messages.map((m) => {
+    const displayTime = formatLocalTime(m.timestamp, timezone);
+    return `<message sender="${escapeXml(m.sender_name)}" time="${escapeXml(displayTime)}">${escapeXml(m.content)}</message>`;
+  });
+
+  const header = `<context timezone="${escapeXml(timezone)}" current_time="${escapeXml(formatCurrentTime(timezone))}" />\n`;
```
Comment (gavrielc) [inline src/router.ts]: Think we can just use formatLocalTime. Also, migt be redundant because we already have the time in the messages

**Comment by TomGranot** [comment]
This builds nicely on the timezone fix from #371. It should fully resolve #483 and #526 (both now closed since the base issue is fixed, but this makes it way more robust with per-group support).

@gavrielc can you give this a look? It touches the agent prompt injection path so want to make sure we're good.
